text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1592 @@
+
+
+### CHANGELOG
+
+```
+feat(cli): user-provided cdk init templates
+```
+
+
+
+### Updated README for Users of CDK CLI
+
+#### `cdk init`
+
+Creates a new CDK project from built-in templates or custom templates. This command helps you get started with AWS CDK by initializing a new project with the necessary files and folders to organize your CDK code, including directories for your application definition, stacks, and constructs.
+
+
+#### Getting Started with Built-in Templates
+
+The CDK provides three built-in templates to help you get started quickly:
+
+1. `app` - CDK Application Template
+
+    * Creates a basic deployable CDK application structure
+    * Includes an empty stack
+    * Available in: TypeScript, JavaScript, Python, Java, C#, F#, Go
+
+2. `lib` - CDK Construct Library Template
+
+    * For building and sharing custom CDK constructs
+    * Available in: TypeScript
+
+3. `sample-app` - Example CDK Application
+
+    * Creates a CDK application with example constructs
+    * Demonstrates common AWS services and patterns
+    * Available in: TypeScript, JavaScript, Python, Java, C#, F#, Go
+
+##### Using Built-In Templates
+
+```
+# Syntax
+# Use a template for a CDK Application
+$ cdk init app --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# Use a template for a CDK Construct Library
+$ cdk init lib --language=typescript
+
+# Use a template for an example CDK Application with some constructs
+$ cdk init sample-app --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# Examples:
+# Create a new TypeScript application
+$ cdk init app --language=typescript
+
+# Create a construct library
+$ cdk init lib --language=typescript
+
+# Create a sample TypeScript application to explore CDK features
+$ cdk init sample-app --language=typescript
+```
+
+
+
+#### Working with Custom Templates
+
+For more advanced use cases, you can run `cdk init` with custom templates from Git repositories or NPM packages to:
+
+* Standardize project structure across your organization
+* Include company-specific configurations and best practices
+* Pre-configure common services and patterns
+
+Let’s walk through an example of running `cdk init` in an empty directory called “example-project” with the custom “my-custom-template” TypeScript template below, which we can pull from a Git repository or NPM package:
+
+```
+my-custom-template/
+└── typescript/
+    ├── package.json               # Required dependency file
+    ├── cdk.template.json          # Required for app templates
+    ├── bin/
+    │   └── app.template.ts        # App entry file
+    ├── lib/
+    │   └── stack.template.ts      # Stack class file
+    ├── test/
+    │   └── stack.test.template.ts # Test file
+    ├── tsconfig.json              # TypeScript configuration
+    ├── .template.gitignore        # Git ignore patterns
+    └── README.md                  # Documentation
```

Comment:
It might be helpful to indicate at a high level that there are only two files that are required inside of `typescript` here, and to be clear that the remaining files are optional. Could be an all-caps comment at the top.